app/parsers/katz.py

```diff
@@ -0,0 +1,133 @@
+from plugins.stockpile.app.parsers.base_parser import BaseParser
+from plugins.stockpile.app.relationship import Relationship
+from app.utility.logger import Logger
+
+import re
+from collections import defaultdict
+
+class MimikatzBlock(object):
+    def __init__(self):
+        self.session = ''
+        self.username = ''
+        self.domain = ''
+        self.logon_server = ''
+        self.logon_time = ''
+        self.sid = ''
+        self.packages = defaultdict(list)
+
+
+class Parser(BaseParser):
+    def __init__(self, parser_info):
+        self.mappers = parser_info['mappers']
+        self.used_facts = parser_info['used_facts']
+        self.parse_mode = 'wdigest'
+        self.log = Logger('parsing_svc')
+
+    def parse_kats(self, output):
+        """
+        Parses mimikatz output with the logonpasswords command and returns a list of dicts of the credentials.
+        Args:
+            output: stdout of "mimikatz.exe privilege::debug sekurlsa::logonpasswords exit"
+        Returns:
+            A list of MimikatzSection objects
+        """
+        # split sections using "Authentication Id" as separator
+        sections = output.split('Authentication Id')
+        creds = []
+        for section in sections:
+            mk_section = MimikatzBlock()
+            package = {}
+            package_name = ''
+            in_header = True
+            pstate = False
+            for line in section.splitlines():
+                line = line.strip()
+                if in_header:
+                    in_header = self._parse_header(line, mk_section)
+                    if in_header:
+                        continue #avoid excess parsing work
+
+                pstate, package_name = self._process_package(line, package, package_name, mk_section)
+                if pstate:
+                    pstate = False
+                    package = {}
+            # save the current ssp if necessary
+            if 'Username' in package and package['Username'] != '(null)' and \
+                    (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                mk_section.packages[package_name].append(package)
+
```

Comment:
Should this be `self._package_extend(package, package_name, mk_section)`

Reply:
Fixed.